### PR TITLE
Cache clients by address only

### LINKED ISSUE #422: Cache pkg/clients per address instead of per (address+key) tuple
Adopt https://github.com/nspcc-dev/neofs-api-go/issues/261 changes.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #422.

Remove both private key and options from cache interface.

### CODE REVIEW

File: pkg/innerring/innerring.go [hunk 1]
```diff
@@ -579,7 +580,7 @@ func createListener(ctx context.Context, p *chainParams) (event.Listener, error)
 	sub, err := subscriber.New(ctx, &subscriber.Params{
 		Log:         p.log,
 		Endpoint:    p.cfg.GetString(p.name + ".endpoint.notification"),
-		DialTimeout: p.cfg.GetDuration(p.name + ".dial_timeouts"),
+		DialTimeout: p.cfg.GetDuration(p.name + ".dial_timeout"),
```
Comment (alexvanin) [inline pkg/innerring/innerring.go:583]: Wow, good one, thanks.

File: pkg/innerring/rpc.go [hunk 2]
```diff
@@ -46,8 +46,10 @@ func newClientCache(p *clientCacheParams) *ClientCache {
 	}
 }
 
-func (c *ClientCache) Get(address string, opts ...client.Option) (*client.Client, error) {
-	return c.cache.Get(c.key, address, opts...)
+func (c *ClientCache) Get(address string) (*client.Client, error) {
+	// Because cache is used by `ClientCache` exclusively,
+	// client will always have valid key.
```
Comment (alexvanin) [inline pkg/innerring/rpc.go:51]: I think you missed `c.getSG`, `c.GetHeader` and `c.GetRangeHash` functions. They use `c.Get` to get client, and then use it without new `WithKey` option.

**Review by alexvanin — COMMENTED** [review]
It seems that key is missing in `getsvc.execCtx` and `searchsvc.execCtx`.

`getsvc.clientCacheWrapper` and `searchsvc.clientCacheWrapper` have `get` function that return `clientWrapper`. This wrapper uses client with options from `execCtx` and it did not adopt new `client.WithKey` option. Can you double check on that?

File: pkg/innerring/rpc.go (hunk 2 above)
Comment (fyrchik) [reply pkg/innerring/rpc.go:51]: Fixed.

**Comment by fyrchik** [comment]
@alexvanin Checked it again:
For both of them, `get` done in `(*execCtx).remoteClient` and corresponding `searchObjects`/`getObject` is executed with the same context (which provide the key from via `callOptions()`).

**Review by alexvanin — APPROVED** [review]
LGTM. I'll try to run it on dev-env to check if everything fine and wait for @cthulhu-rider.

**Review by cthulhu-rider — APPROVED** [review]
The changes are correct, @alexvanin pls merge after testing in dev-env.

**Pull request merged**